python/lsst/ts/ess/csc/data_client/rpi_data_client.py

```diff
@@ -402,10 +406,62 @@ async def write_humidity_etc(
                 location=device_configuration.location,
             )
 
+    async def process_windsonic_telemetry(
+        self,
+        sensor_name: str,
+        timestamp: float,
+        response_code: int,
+        sensor_data: Sequence[float | int],
+    ) -> None:
+        """Process Gill Windsonic sensor telemetry.
+
+        Parameters
+        ----------
+        sensor_name : `str`
+            The name of the sensor.
+        timestamp : `float`
+            The timestamp of the data.
+        response_code : `int`
+            The ResponseCode
+        sensor_data : each of type `float`
+            A Sequence of floats representing the sensor telemetry data:
+
+            * 0: wind speed (m/s)
+            * 1: wind direction (deg)
+        """
+        device_configuration = self.device_configurations[sensor_name]
+
+        if sensor_name not in self.air_flow_cache:
+            self.air_flow_cache[sensor_name] = AirFlowAccumulator(
+                num_samples=device_configuration.num_samples
+            )
+        accumulator = self.air_flow_cache[sensor_name]
+
+        accumulator.add_sample(
+            timestamp=timestamp,
+            speed=sensor_data[0],
+            direction=int(sensor_data[1]),
```

Comment:
Please don't cast to int here. If we need to cast to int to make the XML happy then let's fix the XML (and, as I suggested earlier, do the final casting when writing the event until we do fix the XML).

It's loses even more information to cast to int before computing statistics on the accumulated values.